web_monitoring/db.py

```diff
@@ -1,47 +1,657 @@
+        if page_id is None:
+            url = f'{self._api_url}/versions'
+        else:
+            url = f'{self._api_url}/pages/{page_id}/versions'
+        res = requests.get(url, auth=self._auth, params=params)
+        _process_errors(res)
+        result = res.json()
+        # In place, replace datetime strings with datetime objects.
+        for v in result['data']:
+            v['created_at'] = parse_timestamp(v['created_at'])
+            v['updated_at'] = parse_timestamp(v['updated_at'])
+            v['capture_time'] = parse_timestamp(v['capture_time'])
+        return result
+
+    def get_version(self, version_id):
+        """
+        Lookup a specific Version by ID.
+
+        Parameters
+        ----------
+        version_id : string
+
+        Returns
+        -------
+        response : dict
+        """
+        url = f'{self._api_url}/versions/{version_id}'
+        res = requests.get(url, auth=self._auth)
+        _process_errors(res)
+        result = res.json()
+        data = result['data']
+        data['capture_time'] = parse_timestamp(data['capture_time'])
+        data['updated_at'] = parse_timestamp(data['updated_at'])
+        data['created_at'] = parse_timestamp(data['created_at'])
+        return result
+
+    def add_version(self, page_id, capture_time, uri, hash,
+                    source_type, title, uuid=None, source_metadata=None):
+        """
+        Submit one new Version.
+
+        See :func:`add_versions` for a more efficient bulk importer.
+
+        Parameters
+        ----------
+        page_id : string
+            Page to which the Version is associated
+        uri : string
+            URI of content (such as an S3 bucket or InternetArchive URL)
+        hash : string
+            SHA256 hash of Version content
+        source_type : string
+            such as 'versionista' or 'internetarchive'
+        title : string
+            content of ``<title>`` tag
+        uuid : string, optional
+            A new, unique Version ID (UUID4). If not specified, the server
+            will generate one.
+        source_metadata : dict, optional
+            free-form metadata blob provided by source
+
+        Returns
+        -------
+        response : dict
+        """
+        # Do some type casting here as gentle error-checking.
+        version = _build_version(
+            page_id=page_id,
+            uuid=uuid,
+            capture_time=capture_time,
+            uri=uri,
+            hash=hash,
+            source_type=source_type,
+            title=title,
+            source_metadata=source_metadata)
+        url = f'{self._api_url}/pages/{page_id}/versions'
+        res = requests.post(url, auth=self._auth,
+                            headers={'Content-Type': 'application/json'},
+                            data=json.dumps(version))
+        _process_errors(res)
+        return res.json()
+
+    def add_versions(self, versions):
```

Comment:
There is one more param the API supports; don’t know if we want to add it here: `update`, which can be:
- `skip` (default, leaves already existing versions in place)
- `replace` (replaces existing version with the one in this import)
- `merge` (leaves old fields intact where new values are null and merges `source_metadata` objects).